### PR TITLE
feat: test filter date item

### PR DESCRIPTION
**Pull request by sjaanus**

### CODE REVIEW

File: frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    return recordedChanges;
+};
+
+afterEach(() => {
+    vi.restoreAllMocks();
+});
+
+describe('FilterDateItem Component', () => {
+    it('renders initial state correctly', async () => {
+        vi.mock('hooks/useLocationSettings', () => ({
```
Comment (kwasniew) [inline frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx:32]: import overrides are very hacky and magical as they overwrite how native imports work. Can we handle it similar to how we do local storage?

**Review by kwasniew — APPROVED** [review]
Overall nice change. I'd only try to do it without import mocking.

File: frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx (hunk 1 above)
Comment (sjaanus) [reply frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx:32]: Actually this was a idea I was working on and left in. Removed.

Comment (kwasniew) [reply frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx]: overall I'd recommend not using jest.someMagic or vi.somMagic. It usually means the code is not testable and requires some testing framework magic.

**Pull request merged**